tests/TestAutoType.cpp

```diff
@@ -282,3 +282,30 @@ void TestAutoType::testGlobalAutoTypeRegExp()
     QCOMPARE(m_test->actionChars(), QString("custom_attr_third"));
     m_test->clearActions();
 }
+
+void TestAutoType::testAutoTypeSyntaxChecks()
+{
+    // Huge sequence
+    QCOMPARE(true, AutoType::checkSyntax("{word 23}{F1 23}{~ 23}{% 23}{^}{F12}{(}{) 23}{[}{[}{]}{Delay=23}{+}{-}~+%@fixedstring"));
```

Comment:
@TheZ3ro I just noticed there's no support for `{[}` and `{]}`, can you add it along with these changes or would  be better to open a separate PR?

Reply:
Also, `{-}` should trigger a syntax error, since the [correct code is `{SUBSTRACT}`](https://keepass.info/help/base/autotype.html)

Reply:
Why `{word}` should trigger an error? Because it's not a supported literal ?
Sadly RN we don't have a filter on supported literals so unless it has a different scheme (like `{BEEP X Y}`) we have to blindly accept it, then the action parser will skip unsupported ones

Reply:
For the `{[}` literals better make a new PR, I will work on that after merging this old one

Reply:
>then the action parser will skip unsupported ones

Seems fair, since it won't break the entire sequence